src/components/clusterexplorer/explorer.ts

```diff
@@ -77,12 +79,20 @@ export function isKubernetesExplorerResourceNode(obj: any): obj is ClusterExplor
     return obj && obj.nodeCategory === KUBERNETES_EXPLORER_NODE_CATEGORY && obj.nodeType === 'resource';
 }
 
+export interface TreeViewNodeStateChangeEvent<T> extends vscode.TreeViewExpansionEvent<T> {
+	state: vscode.TreeItemCollapsibleState;
+}
+
 export class KubernetesExplorer implements vscode.TreeDataProvider<ClusterExplorerNode> {
     private onDidChangeTreeDataEmitter: vscode.EventEmitter<ClusterExplorerNode | undefined> = new vscode.EventEmitter<ClusterExplorerNode | undefined>();
     readonly onDidChangeTreeData: vscode.Event<ClusterExplorerNode | undefined> = this.onDidChangeTreeDataEmitter.event;
 
     private readonly extenders = Array.of<ExplorerExtender<ClusterExplorerNode>>();
     private readonly customisers = Array.of<ExplorerUICustomizer<ClusterExplorerNode>>();
+    private viewer: vscode.TreeView<ClusterExplorerNode>;
```

Comment:
Yes for the local variable. Not sure about the `create()`. `Create()` returns an instance of KubernetesExplorer but the `initialize()` a set of disposables so that the extension can dispose them when deactivated, so not sure it's a good idea to mix them. Leave this to you :)